.github/actions/pre-commit/action.yml

```
@@ -29,6 +37,18 @@ runs:
         ${{ inputs.pre-commit-all-files }} && EXTRA_ARGS='--all-files'
         EXTRA_ARGS="${EXTRA_ARGS} ${{ inputs.pre-commit-args }}"
         echo EXTRA_ARGS="${EXTRA_ARGS}" >> $GITHUB_ENV
+
     - uses: pre-commit/action@v3.0.0
+      id: pre-commit
+      continue-on-error: true
       with:
         extra_args: ${{ env.EXTRA_ARGS }}
+
+    - uses: stefanzweifel/git-auto-commit-action@v4
+      id: auto-commit-action
+      if: inputs.auto-commit == 'true'
+
+    - name: rethrow failure when autocommit is disable or no changes has been detected
+      if: steps.pre-commit.outcome == 'failure' && (inputs.auto-commit != 'true' || steps.auto-commit-action.outputs.changes_detected == 'false')
+      run: echo "pre-commit failed, auto-commit is disabled or no changes detected for auto-commit" && exit 1
```

(if continue-on-error suggestion is valid and done)

```suggestion
      run: echo "pre-commit failed and no changes detected for auto-commit" && exit 1
```